packages/frontend/src/hooks/dashboard/useDashboardFiltersForExplore.ts

```diff
@@ -1,26 +1,63 @@
-import { DashboardFilters, Explore } from '@lightdash/common';
-import { useMemo } from 'react';
+import {
+    DashboardFilterRule,
+    DashboardFilters,
+    Explore,
+} from '@lightdash/common';
+import { useCallback, useMemo } from 'react';
 import { useDashboardContext } from '../../providers/DashboardProvider';
 
 const useDashboardFiltersForExplore = (
+    tileUuid: string,
     explore: Explore | undefined,
 ): DashboardFilters => {
     const { dashboardFilters, dashboardTemporaryFilters } =
         useDashboardContext();
 
+    const tables = useMemo(
+        () => (explore ? Object.keys(explore.tables) : []),
+        [explore],
+    );
+
+    const overrideTileFilters = useCallback(
```

Comment:
I think this logic would also be simpler if the `tileTargetOverride` was a map.

It feels like the `DashboardFilterRule` has a dual purpose, it holds the configuration for the dashboard filters but then also the filters themselves.

i.e. the moment where we go:

```
...f,
target: {
  fieldId: tileConfig.fieldId,
  tableName: tileConfig.tableName
}
```

This still has the `tileTargetOverride` property, which looks a bit strange.


I feel like this function should take the configuration (includes tile overrides) and spit out a set of valid filters (not knowing anything about tile overrides)

Reply:
What is the type returned by this callback?